### DISCUSSION TITLE
Pass CancellationToken to AsyncCommand.ExecuteAsync()?

### DISCUSSION THREAD

**Discussion by wjrogers** [spectreconsole/spectre.console#659] [Ideas]
I gave Spectre.Console.Cli a spin by porting an existing project, and one of the pain points was co-operative cancellation of asynchronous commands. The library I had been using previously passed a `CancellationToken` to its abstract "execute" method, so I had a bunch of existing code that expected to receive one. I had to resort to constructor injection of `IHostApplicationLifetime` to get its `StoppingToken` in every command, which is a lot of boilerplate when you have more than a trivial number of commands.

It seems like Spectre could add a `CommandContext.CancellationToken` property without breaking its existing API. By default, this could be `CancellationToken.None`. Then, an overload could be added to `CommandApp.RunAsync()` that accepts a `CancellationToken` and propagates it to the `CommandContext`. (Its `CanBeCanceled` property would allow implementers to discover whether cancellation is "wired up" or not.)

Would you be interested in this feature? I think I could manage a pull request that adds it.

**Comment by nils-a** [comment]
#701 is probably related to this
